### PR TITLE
fix/mutable_config

### PR DESCRIPTION
**Pull request by JarbasAl**
makes `self.config_core` a property, ensures we always get the latest mycroft.conf

`Configuration` is a singleton, keeps an internal cache and reacts to bus events. This does not repeatedly read from disk

this is particularly relevant in the `self.location` property, once PHAL is implemented it will also solve the associated TODO

it is also relevant for `self.lang`, the hypothetical future algorithm to switch mycroft language on the fly will need this

relevant snippet from `MycroftSkill` class
```python
  @property
    def lang(self):
        """Get the current language."""
        lang = self._core_lang
        message = dig_for_message()
        if message:
            lang = message.data.get("lang") or lang
        return lang.lower()

    @property
    def _core_lang(self):
        """Get the configured default language.
        NOTE: this should be public, but since if a skill uses this it wont
        work in regular mycroft-core it was made private! Equivalent PRs in
        mycroft-core have been rejected/abandoned"""
        return self.config_core.get("lang", "en-us").lower()
        
    @property
    def location(self):
        """Get the JSON data struction holding location information."""
        # TODO: Allow Enclosure to override this for devices that
        # contain a GPS.
        return self.config_core.get('location')

    @property
    def location_pretty(self):
        """Get a more 'human' version of the location as a string."""
        loc = self.location
        if type(loc) is dict and loc['city']:
            return loc['city']['name']
        return None

    @property
    def location_timezone(self):
        """Get the timezone code, such as 'America/Los_Angeles'"""
        loc = self.location
        if type(loc) is dict and loc['timezone']:
            return loc['timezone']['code']
        return None
```

related issues:
- https://github.com/MycroftAI/mycroft-core/issues/2859

### CODE REVIEW

**Comment by JarbasAl** [comment]
this one stays on hold, testing in alpha release revealed that some downstreams are assigning to some variables that become properties with this PR. this is a breaking change that warrants further discussion. 

I still think this is a good change, but if its breaking then maybe its also a chance to further improve some things so it is only a breaking change once....

File: mycroft/client/enclosure/base.py [hunk 1]
```diff
@@ -20,15 +20,21 @@
...
-        # Load full config
-        config = Configuration.get()
-        self.lang = config['lang']
-        self.config = config.get("enclosure")
-        self.global_config = config
-
         # Create Message Bus Client
         self.bus = MessageBusClient()
 
+    @property
```
Comment (NeonDaniel) [inline mycroft/client/enclosure/base.py:26]: Same as above; these were exposed variables, so custom enclosures may have overridden them

**Review by NeonDaniel — COMMENTED** [review]
These changes will require some manual testing on my part (probably after 0.0.2 release). This breaks code that imports and overrides any of these variables that are now properties; following semver, these changes would constitute ovos-core 1.0.0

**Comment by ChanceNCounter** [comment]
I will also want to toy with them, but for the other reason! I am very excited about these functions 🤓

**Comment by JarbasAl** [comment]
replaced by #105

**Pull request closed without merging**